pilot/pkg/config/file/store.go

```diff
@@ -78,7 +81,20 @@ func (s *KubeSource) Get(typ config.GroupVersionKind, name, namespace string) *c
 }
 
 func (s *KubeSource) List(typ config.GroupVersionKind, namespace string) ([]config.Config, error) {
-	return s.inner.List(typ, namespace)
+	configs, err := s.inner.List(typ, namespace)
+	if err != nil {
+		return nil, err
+	}
+	if namespace == model.NamespaceAll && s.namespacesFilter != nil {
+		var out []config.Config
+		for _, config := range configs {
+			if s.namespacesFilter(config) {
```

Comment:
Does this scale...? List() is processing a LOT of items - it was a bottleneck in the code prior to some performance improvements. `Filter` uses a lock on each call - we may end up needing to acquire 10k locks for 1 call to List potentially.

The current code is passing the filter to NewFilteredSharedIndexInformer, instead - this is very different since it is only processing each event, not all items on every call to List